cpp/csp/cppnodes/statsimpl.h

```diff
@@ -1214,52 +1221,50 @@ class Quantile
                     break;
             }
         #else
-            auto it = m_tree.begin();
-            std::advance( it, ft );
-            switch ( m_interpolation )
+            switch (m_interpolation)
             {
-                case LINEAR:
-                    if( ft == target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        double lower = *it;
-                        double higher = *++it;
-                        qtl = ( 1 - target + ft ) * lower + ( 1 - ct + target ) * higher;
-                    }
-                    break;
-                case LOWER:
-                    qtl = *it;
-                    break;
-                case HIGHER:
-                    qtl = ( ft == ct ? *it : *++it );
-                    break;
-                case MIDPOINT:
-                    if( ft == target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        double lower = *it;
-                        double higher = *++it;
-                        qtl = ( higher+lower ) / 2;
-                    }
-                    break;
-                case NEAREST:
-                    if( target - ft <= ct - target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        qtl = *++it;
-                    }
-                    break;
-                default:
-                    break;
+            case LINEAR:
+                if (ft == target)
+                {
+                    qtl = *m_tree.get<0>().nth(ft);
```

Comment:
You can call `auto fIt = m_tree.get<0>().nth( ft )` before the switch statement and it will simplify the code.
Then you can access the floor value as `*fIt ` and the ceil value as `*++fIt `.

Reply:
I ended up adding a ternary as *++fIt doesn't work if ft == ct, it will iterate when there shouldn't be an iteration.